layer_metadata_restorer_test.go

```diff
@@ -28,19 +28,24 @@ func testLayerMetadataRestorer(t *testing.T, when spec.G, it spec.S) {
 	var (
 		layerDir       string
 		layerAnalyzer  lifecycle.LayerMetadataRestorer
+		layerSHAStore  lifecycle.LayerSHAStore
 		layersMetadata platform.LayersMetadata
 		cacheMetadata  platform.CacheMetadata
 		buildpacks     []buildpack.GroupBuildpack
 		skipLayers     bool
+		useShaFiles    bool
+		logger         log.Logger
 	)
 
 	it.Before(func() {
 		var err error
 
 		layerDir, err = ioutil.TempDir("", "lifecycle-layer-dir")
 		h.AssertNil(t, err)
-		logger := &log.Logger{Handler: &discard.Handler{}}
-		layerAnalyzer = lifecycle.NewLayerMetadataRestorer(logger, layerDir, skipLayers)
+		useShaFiles = true // notice - the default for platform API >= 0.7 is false (it's set to false in some of the tests)
+		logger = log.Logger{Handler: &discard.Handler{}}
+		layerAnalyzer = lifecycle.NewLayerMetadataRestorer(&logger, layerDir, skipLayers, useShaFiles)
+		layerSHAStore = lifecycle.NewLayerSHAStore(layerAnalyzer.UseSHAFiles())
```

Comment:
```suggestion
		layerSHAStore = lifecycle.NewLayerSHAStore(useShaFiles)
```

Reply:
I know that it's possible to make this change, but don't you think it's clearer the to keep it as is because this is how we do it in the real code?
This is how we keep both values sync for sure.

Reply:
Eh, I don't feel strongly about it. If you do, keep it as is 😄

Reply:
I also don't have strong opinion about it. Just trying to make our code as clear as possible to everyone 😊
@natalieparellano, what do you think?

Reply:
See https://github.com/buildpacks/lifecycle/commit/f584e397d9e441ee1860957da6c18e023d65a89d#r51743291 (idea for making the layer SHA store "dumber")

Reply:
Marking this conversation as resolved following https://github.com/buildpacks/lifecycle/pull/624#commitcomment-51744864.

Reply:
If we do https://github.com/buildpacks/lifecycle/pull/624/files#r645837750 this question goes away.

Reply:
(nit) This wasn't introduced in this PR, but could we change the name of `layerAnalyzer` in the test file to `layerMetadataRestorer` (or some abbreviated version of it)? `layerAnalyzer` is a hold out from when a previous PR was architected differently.

Reply:
Renamed `layerAnalyzer` to `layerMetadataRestorer` in https://github.com/buildpacks/lifecycle/pull/624/commits/debb186ed16488b89d25430039aa293877d0f6c8.
Going to mark this conversation as resolved and to track the changes that relates to the `useShaFile` boolean as part of https://github.com/buildpacks/lifecycle/pull/624/files#r645837750.